crates/dojo-core/src/world.cairo

```diff
@@ -564,6 +565,45 @@ mod world {
         }
     }
 
+    #[abi(embed_v0)]
+    impl UpgradeableState of IUpgradeableState<ContractState> {
+        fn upgrade_state(ref self: ContractState, new_state: Span<felt252>, program_output: Span<felt252>) {
```

Comment:
Let's use Cairo 1 serialization here if you agree. :)

```rust
struct StorageUpdate {
    key: felt252,
    value: felt252,
}

// Span<StorageUpdate>
```
Then, for the program output, we should also use a struct as we should already have it.

Reply:
hashing structs makes it more complex than hashing spans, take a look [here](https://github.com/dojoengine/dojo/pull/1834/commits/1f5e1d4c5c7a5c65c07c6585f373343b9f930367) and lmk what do you think